### PR TITLE
Fix #3537: merge hints summary xml files into one file

### LINKED ISSUE #3537: Merge hints_summary.xml into single xml file
Currently there are 4 versions of `hints_summary.xml` file merge it into single xml file.

We can use https://text-compare.com/ to compare two versions of this file and for all the differences we can create variables in `dimens.xml` file and use it accordingly.

It can happen that it becomes really hard to merge all 4 files into single file, in such cases prefer code-readability over single file. Therefore it can be fine if in the end the code still contains 2 or 3 files.

**Note**: **In PR, make sure you add before and after screenshot of mobile-portrait, mobile-landscape, tablet-portrait and tablet-landscape for comparison and make sure that there is not difference between before and after UI.**

### PR DESCRIPTION
**Pull request by ma-za-kpe**
## Explanation
This pr merges all four hint_summary.xml files into on existing default XML file. It uses dimen files to set different dimensions for individual device configuations. 

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only

If your PR includes UI-related changes, then:
- Add screenshots for portrait/landscape for both a tablet & phone of the before & after UI changes
- For the screenshots above, include both English and pseudo-localized (RTL) screenshots (see [RTL guide](https://github.com/oppia/oppia-android/wiki/RTL-Guidelines))
- Add a video showing the full UX flow with a screen reader enabled (see [accessibility guide](https://github.com/oppia/oppia-android/wiki/Accessibility-(A11y)-Guide))
- Add a screenshot demonstrating that you ran affected Espresso tests locally & that they're passing

## BEFORE SCREENSHOTS
tab_land_hint
tab_port_hint
phone_land_hint
phone_port_hint
phone_port_brfore
phone_land_before
tab_port_before
tab_land_before

 
## AFTER SCREENSHOTS

tab_port_hint_after
phone_port_hint_after
phone_land_hint_after
tab_land_hint_after

tab_land_after
tab_port_after
phone_port_after
phone_land_after

### CODE REVIEW

File: app/src/main/res/layout/hints_summary.xml [hunk 1]
```diff
@@ -19,23 +19,23 @@
...
 
       <TextView
         android:id="@+id/hint_title"
         style="@style/TextViewStart"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:layout_marginTop="12dp"
-        android:layout_marginBottom="12dp"
+        android:layout_marginTop="@dimen/hint_title_margin_top"
+        android:layout_marginBottom="@dimen/hint_title_margin_bottom"
```
Comment (vrajdesai78) [inline app/src/main/res/layout/hints_summary.xml]: Also add android:paddingBottom and assign its dimensions in dimens.xml files

File: app/src/main/res/values-land/dimens.xml [hunk 2]
```diff
@@ -480,7 +480,29 @@
...
   <dimen name="hints_and_solution_recycler_view_padding_top">24dp</dimen>
   <dimen name="hints_and_solution_recycler_view_padding_bottom">120dp</dimen>
-  
+
+  <!-- Hints summary  -->
+  <dimen name="hints_summary_linear_layout_margin_top">32dp</dimen>
+  <dimen name="hints_summary_constraint_layout_layout_margin_start">72dp</dimen>
+  <dimen name="hints_summary_constraint_layout_layout_margin_end">72dp</dimen>
+  <dimen name="hints_summary_constraint_layout_layout_margin_bottom">20dp</dimen>
+
```
Comment (vrajdesai78) [inline app/src/main/res/values-land/dimens.xml]: remove this extra space

File: app/src/main/res/values-land/dimens.xml [hunk 3]
```diff
@@ -480,7 +480,29 @@
...
+  <dimen name="hint_list_container_layout_margin_start">84dp</dimen>
+  <dimen name="hint_list_container_layout_margin_top">8dp</dimen>
+  <dimen name="hint_list_container_layout_margin_end">84dp</dimen>
+  <dimen name="hint_list_container_layout_margin_bottom">32dp</dimen>
+
```
Comment (vrajdesai78) [inline app/src/main/res/values-land/dimens.xml]: remove this extra space

File: app/src/main/res/values-land/dimens.xml [hunk 4]
```diff
@@ -480,7 +480,29 @@
...
+  <dimen name="hint_title_margin_top">12dp</dimen>
+  <dimen name="hint_title_margin_bottom">12dp</dimen>
+
```
Comment (vrajdesai78) [inline app/src/main/res/values-land/dimens.xml]: remove this extra space

File: app/src/main/res/values-land/dimens.xml [hunk 5]
```diff
@@ -480,7 +480,29 @@
...
+  <dimen name="reveal_hint_btn_margin_top">16dp</dimen>
+  <dimen name="reveal_hint_btn_margin_bottom">32dp</dimen>
+
```
Comment (vrajdesai78) [inline app/src/main/res/values-land/dimens.xml]: remove this extra space

File: app/src/main/res/values-sw600dp-land/dimens.xml [hunk 6]
```diff
@@ -468,6 +468,24 @@
   <dimen name="hints_and_solution_recycler_view_padding_top">4dp</dimen>
   <dimen name="hints_and_solution_recycler_view_padding_bottom">112dp</dimen>
 
+  <!-- Hints summary  -->
+  <dimen name="hints_summary_constraint_layout_layout_margin_start">176dp</dimen>
+  <dimen name="hints_summary_constraint_layout_layout_margin_end">176dp</dimen>
+  <dimen name="hints_summary_constraint_layout_layout_margin_bottom">4dp</dimen>
+
```
Comment (vrajdesai78) [inline app/src/main/res/values-sw600dp-land/dimens.xml]: Here and everywhere else where you have given extra space in dimens.xml remove it as mentioned above

File: app/src/main/res/values-land/dimens.xml [hunk 7]
```diff
@@ -480,7 +480,29 @@
   <!-- Hints and Solution Fragment -->
   <dimen name="hints_and_solution_recycler_view_padding_top">24dp</dimen>
   <dimen name="hints_and_solution_recycler_view_padding_bottom">120dp</dimen>
-  
+
+  <!-- Hints summary  -->
```
Comment (vrajdesai78) [inline app/src/main/res/values-land/dimens.xml]: change 'summary' to 'Summary' here and in other dimens.xml also

File: app/src/main/res/values-sw600dp-port/dimens.xml [hunk 8]
```diff
@@ -467,6 +467,23 @@
   <dimen name="hints_and_solution_recycler_view_padding_top">0dp</dimen>
   <dimen name="hints_and_solution_recycler_view_padding_bottom">112dp</dimen>
 
+  <!-- Hints summary  -->
+  <dimen name="hints_summary_constraint_layout_layout_margin_start">128dp</dimen>
+  <dimen name="hints_summary_constraint_layout_layout_margin_end">128dp</dimen>
+  <dimen name="hints_summary_constraint_layout_layout_margin_bottom">4dp</dimen>
+
+  <dimen name="hint_list_container_layout_margin_start">128dp</dimen>
```
Comment (vrajdesai78) [inline app/src/main/res/values-sw600dp-port/dimens.xml]: Name for every dimension should start from name of activity/fragment. So, here and everywhere else start the name of dimension with hints_summary as you have done for previous dimensions. Here, change the current name `hint_list_container_layout_margin_start` to `hints_summary_hint_list_container_layout_margin_start`

File: app/src/main/res/values-sw600dp-port/dimens.xml [hunk 9]
```diff
@@ -467,6 +467,23 @@
...
+  <dimen name="hint_list_container_layout_margin_end">128dp</dimen>
+  <dimen name="hint_list_container_layout_margin_bottom">32dp</dimen>
+
+  <dimen name="hint_title_margin_top">44dp</dimen>
+  <dimen name="hint_title_padding_bottom">28dp</dimen>
+
+  <dimen name="reveal_hint_btn_margin_bottom">28dp</dimen>
```
Comment (vrajdesai78) [inline app/src/main/res/values-sw600dp-port/dimens.xml]: Don't use abbreviations like btn, use full name button

File: app/src/main/res/values-sw600dp-port/dimens.xml [hunk 10]
```diff
@@ -467,6 +467,23 @@
...
+
+  <dimen name="expand_list_icon_margin_top">32dp</dimen>
```
Comment (vrajdesai78) [inline app/src/main/res/values-sw600dp-port/dimens.xml]: To maintain consistency, make sure that all dimensions are present in all four dimens.xml files. For example, here you mentioned just two dimensions for `expand_list_icon*`. So, add another two dimensions i.e. `hints_summary_expand_list_icon_margin_end` and `hints_summary_expand_list_icon_margin_start` and assign 0 dp. Do these for other dimensions also

**Review by vrajdesai78 — CHANGES_REQUESTED** [review]

**Comment by ma-za-kpe** [comment]
Hello @vrajdesai78 , i have made the requested changes, please have a look.

File: app/src/main/res/layout/hints_summary.xml [hunk 11]
```diff
@@ -19,23 +19,24 @@
...
+        android:paddingBottom="@dimen/hint_title_padding_bottom"
```
Comment (vrajdesai78) [inline app/src/main/res/layout/hints_summary.xml]: Here, and in other places also you have to change the dimen name exactly same as you mentioned in dimens.xml. Here, you have written `@dimen/hint_title_padding_bottom` which should be `@dimen/hints_summary_hint_title_padding_bottom`

File: app/src/main/res/layout/hints_summary.xml [hunk 12]
```diff
@@ -47,9 +48,9 @@
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
         android:layout_marginStart="8dp"
-        android:layout_marginTop="16dp"
+        android:layout_marginTop="@dimen/reveal_hint_btn_margin_top"
```
Comment (vrajdesai78) [inline app/src/main/res/layout/hints_summary.xml]: change dimen name

File: app/src/main/res/layout/hints_summary.xml [hunk 13]
```diff
@@ -47,9 +48,9 @@
...
         android:layout_marginEnd="8dp"
-        android:layout_marginBottom="32dp"
+        android:layout_marginBottom="@dimen/reveal_hint_btn_margin_bottom"
```
Comment (vrajdesai78) [inline app/src/main/res/layout/hints_summary.xml]: change dimen name

File: app/src/main/res/layout/hints_summary.xml [hunk 14]
```diff
@@ -66,9 +67,10 @@
         android:id="@+id/expand_list_icon"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:layout_marginStart="8dp"
-        android:layout_marginEnd="8dp"
-        android:layout_marginBottom="8dp"
+        android:layout_marginStart="@dimen/expand_list_icon_margin_start"
```
Comment (vrajdesai78) [inline app/src/main/res/layout/hints_summary.xml]: change dimen name

File: app/src/main/res/layout/hints_summary.xml [hunk 15]
```diff
@@ -95,10 +97,10 @@
       android:id="@+id/hint_list_container"
       android:layout_width="match_parent"
       android:layout_height="wrap_content"
-      android:layout_marginStart="40dp"
-      android:layout_marginTop="8dp"
-      android:layout_marginEnd="40dp"
-      android:layout_marginBottom="32dp"
+      android:layout_marginStart="@dimen/hint_list_container_layout_margin_start"
```
Comment (vrajdesai78) [inline app/src/main/res/layout/hints_summary.xml]: change dimen name

**Review by vrajdesai78 — CHANGES_REQUESTED** [review]
After doing changes, run your app again and update the screenshots.

**Comment by ma-za-kpe** [comment]
Okay, let me upload new screenshots...

**Comment by ma-za-kpe** [comment]
Hello @vrajdesai78, I have made the requested changes, with new screenshots

**Review by vrajdesai78 — APPROVED** [review]

**Comment by rt4914** [comment]
@vrajdesai78 @ma-za-kpe  In the PR description the screenshots are not correct.

Those screenshots have been taken from android studio where they show the previews which might be incorrect. Instead you should create screenshot in devices mainly mobile-portrait, mobile-landscape, tablet-portrait, tablet-landscape.

Also each screenshot should show atleast 2 hints where one is open and another is closed.

@vrajdesai78 Can you help @ma-za-kpe in creating such screenshots.

**Comment by ma-za-kpe** [comment]
Hello, I have updated the screenshots as required.

**Comment by ma-za-kpe** [comment]
Hello, I have added the other variation of screenshots. @vrajdesai78

**Pull request closed without merging**